src/main/scala/Battle.scala

```diff
@@ -0,0 +1,136 @@
+import scala.util.Random
+import scala.io.StdIn.{readLine,readInt}
+
+class Battle{
+  val r: Random.type = scala.util.Random
+  def action(){
```

Comment:
this method could be split into two methods one being a helper, the first part being the printlines and the second being the reading input and matching against it

also I think you don't need val choice and you should be able to directly do

readInt match {